src/python/bcc/__init__.py

```diff
@@ -1366,6 +1366,7 @@ def cleanup(self):
             self.tracefile = None
         for name, fn in list(self.funcs.items()):
             os.close(fn.fd)
+            del self.funcs[name]
```

Comment:
Instead of deleting each individual element, may be clear thee whole directory after the `for` loop with `self.funcs.clear()`?

Reply:
Rational was to keep `self` in a coherent state, where each **successful** `os.close()` call is followed by a cleanup of the associated item of `funcs` - a pattern exhibited in other parts of `self.cleanup()`.

Reply:
okay, we can keep it consistent with other usage for now.